This submission can get you up to 5 extra points on your midterm. It is due November 6th at 6:30 PM sharp - nothing submitted after that will be accepted. Please submit it to my email with the subject 'Extra Credit Assignment \#1'

In [2]:
import scipy.stats as stats
import numpy as np

1. Say that we flip 10 coins 100 times. What is the expected number of times we will see 0 - 10 heads in our simulation? (i.e. we'll see 5 heads 100 times, or 0 heads 20 times and 10 heads 80 times)

In [4]:
np.random.seed(42)
dist = stats.binom(n=10, p=0.5)
results = dist.rvs(size=100)
for i in range(11):
    print("We will see", i, "heads", int(round(dist.pmf(i)*100)), "times")

We will see 0 heads 0 times
We will see 1 heads 1 times
We will see 2 heads 4 times
We will see 3 heads 12 times
We will see 4 heads 21 times
We will see 5 heads 25 times
We will see 6 heads 21 times
We will see 7 heads 12 times
We will see 8 heads 4 times
We will see 9 heads 1 times
We will see 10 heads 0 times


2. Now say I have the below observation. At a signifiance level of 0.05, can I reject the null hypothesis that the observation comes from the expected distribution?

In [7]:
stats.chi2(df=9).ppf(0.95)

16.918977604620448

In [8]:
1 - stats.chi2(df=10).cdf(29.702)

0.0009582277855237908

The p-value for our Chi-square statistic is very close to 0, meaning we can reject the null hypothesis at a significance level of 0.05.

3. At what value can I reject the null hypothesis?

If the value is greater than 16.9 we can reject the null hypothesis.